pkg/reconciler/instances/istio/action.go

```diff
@@ -1,37 +1,115 @@
 package istio
 
 import (
+	"bytes"
 	"fmt"
 	"github.com/kyma-incubator/reconciler/pkg/reconciler"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/chart"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/file"
 	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
 	"github.com/pkg/errors"
+	"io"
+	"log"
+	"os"
 	"os/exec"
+	"strings"
 )
 
 const (
-	istioctl1_10_2 = "/bin/istioctl-1.10.2"
+	istioctl1_11_1      = "/bin/istioctl-1.11.1" // change path to the local `istioctl` if debugging locally
+	yaml_delimiter      = "---"
+	istio_operator_kind = "kind: IstioOperator"
 )
 
 type ReconcileAction struct {
 }
 
 func (a *ReconcileAction) Run(version, profile string, config []reconciler.Configuration, context *service.ActionContext) error {
-	ws, err := context.WorkspaceFactory.Get(version)
+	var overrides = make(map[string]interface{}, len(config))
+
+	for _, configEntry := range config {
+		overrides[configEntry.Key] = configEntry.Value
+	}
+
+	comp := chart.NewComponent("istio-configuration", "istio-system", overrides)
+
+	componentSet := chart.NewComponentSet(context.KubeClient.Kubeconfig(), version, profile, []*chart.Component{comp})
+	log.Println("Before rendering the manifest")
+
+	manifests, err := context.ChartProvider.Manifests(componentSet, false, &chart.Options{})
+	if err != nil {
+		return err
+	}
+
+	manifestsCount := len(manifests)
+	if manifestsCount != 1 {
+		return errors.Errorf("One manifest expected but got %d", manifestsCount)
+	}
+
+	finalManifest := manifests[0].Manifest
+
+	istioOperator := extractIstioOperatorContextFrom(finalManifest)
+	istioOperatorPath, istioOperatorCf, err := file.Path(istioOperator)
+	if err != nil {
+		return err
+	}
+
+	context.Logger.Infof(istioOperatorPath)
+
+	defer func() {
+		cleanupErr := istioOperatorCf()
+		if cleanupErr != nil {
+			context.Logger.Error(cleanupErr)
+		}
+	}()
+
+	kubeconfiigPath, kubeconfigCf, err := file.Path(context.KubeClient.Kubeconfig())
+	if err != nil {
+		return err
+	}
+
+	context.Logger.Infof(kubeconfiigPath)
+
+	defer func() {
+		cleanupErr := kubeconfigCf()
+		if cleanupErr != nil {
+			context.Logger.Error(cleanupErr)
+		}
+	}()
+
+	cmd := prepareIstioctlCommand(istioOperatorPath, kubeconfiigPath)
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+
+	_, err = context.KubeClient.Deploy(nil, finalManifest, "istio-system", nil)
```

Comment:
What is the purpose of this command?

Reply:
It is used to apply the rest of the resources to the cluster. Have a look at `Istio-configuration` component and what it  delivers (it is not only Istio Operator itself).

Reply:
Ok. Clear.